snappiershot/serializers/json.py

```diff
@@ -75,10 +78,24 @@ def default(self, value: Any) -> Any:
         if isinstance(value, DATETIME_TYPES):
             return self.encode_datetime(value)
 
+        if self._is_pandas_object(value):
+            return self.encode_pandas(value)
+
         raise NotImplementedError(  # pragma: no cover
             f"Encoding for this object is not yet implemented: {value} ({type(value)})"
         )
 
+    @staticmethod
+    def _is_pandas_object(obj: object) -> bool:
+        """Return true if the given object is a pandas object
+
+        A special effort is made to avoid importing pandas unless it's really necessary.
+        """
+        pd = get_pandas()
+        if pd is not None:
+            return isinstance(obj, PANDAS_TYPES)
```

Comment:
do we want to only check against our types here, or do we want to check if an objects type is from the pandas module?

Reply:
change, you may want to review the new approach